packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts

```diff
@@ -144,11 +151,59 @@ export async function generateSpokenMathDetails(mathString: string) {
     // to look for individual math blocks.
     if (firstSection.type === "paragraph") {
         for (const piece of firstSection.content) {
-            piece.type === "math"
-                ? (convertedSpeech += engine.texToSpeech(piece.content))
-                : (convertedSpeech += piece.content);
+            switch (piece.type) {
+                case "math":
+                    convertedSpeech += engine.texToSpeech(piece.content);
+                    break;
+                case "unescapedDollar":
+                    // If the unescaped dollar had a closing pair to define
+                    // math, it would have been caught by the "math" case above.
+                    // Since this unescaped dollar is caught here, we can
+                    // assume it is alone and used as as a literal dollar sign.
+                    convertedSpeech += "$";
+                    break;
+                default:
+                    convertedSpeech += piece.content;
+                    break;
+            }
         }
     }
 
     return convertedSpeech;
 }
+
+/**
+ * Take an array of LockedLabelType object and joins the text of each label
+ * with a comma and space in between. The text of each label is converted to
+ * spoken math using the SpeechRuleEngine.
+ */
+export async function joinLabelsAsSpokenMath(
+    labels: LockedLabelType[] | undefined,
+): Promise<string> {
+    if (!labels || labels.length === 0) {
+        return "";
+    }
+
+    const spokenLabelPromises = labels.map(async (label) => {
+        return await generateSpokenMathDetails(label.text);
+    });
```

Comment:
Minor: I don't think you need async/await here. You can just return the promise.

```suggestion
    const spokenLabelPromises = labels.map((label) => {
        return generateSpokenMathDetails(label.text);
    });
```